library/L2_HAL/communication/test/tsop752_test.cpp

```diff
@@ -0,0 +1,143 @@
+#include "L2_HAL/communication/tsop752.hpp"
+#include "L4_Testing/testing_frameworks.hpp"
+
+namespace sjsu
+{
+EMIT_ALL_METHODS(Tsop752);
+
+TEST_CASE("Tsop752 Infrared Receiver Test", "[tsop752]")
+{
+  Mock<PulseCapture> mock_pulse_capture;
+  Mock<Timer> mock_timer;
+
+  Fake(Method(mock_pulse_capture, ConfigureCapture),
+       Method(mock_pulse_capture, EnableCaptureInterrupt));
+  Fake(Method(mock_timer, SetMatchBehavior),
+       Method(mock_timer, Start),
+       Method(mock_timer, Stop),
+       Method(mock_timer, Reset));
+
+  const Status kFailureStatus[] = {
```

Comment:
```suggestion
  const std::array kFailureStatus[] = {
```
Prefer to use std::array vs c-style arrays.